recipe/menuinst_patches/0001-Do-not-assume-menuinst-wants-to-operate-on-sys.prefi.patch

```diff
@@ -29,16 +63,21 @@ index a21240a..1d14731 100644
 -                                              path, bool(remove), prefix, True)])
 -                else:
 -                    retcode = 1
-+                    if exists(join(prefix, 'python.exe')):
-+                        retcode = runAsAdmin([join(prefix, 'python.exe'), '-c',
-+                            "import menuinst; menuinst.install(%r, %r, %r, %r)" % (
-+                                path, bool(remove), prefix, True)])
++                    retcode = runAsAdmin([join(root_prefix, 'python'), '-c',
++                                          "import menuinst; menuinst.install(%r, %r, %r, %r, %r)" % (
++                                              path, bool(remove), prefix, True, root_prefix)])
```

Comment:
Thank you for all the work here, @ericpre !

Doesn't this call to `runAsAdmin()` assume that (installed) menuinst supports the `root_prefix` argument? It is going to subprocess a call to python and not use the module shipped within conda-standalone for that particular call.

Reply:
Thanks for the review @nehaljwani, this is a good question, which made me doubt for a bit!
The way the code is written is a bit misleading and in the context of `conda-standalone`, the answer is actually simple: this code is not called for admin install, because it uses the [`_install` function in the if block (`if isUserAdmin()`) ](https://github.com/conda/menuinst/blob/e42209ad29f432a8148bff7935932d2841c283e0/menuinst/__init__.py#L60-L61).

I have checked running the installer built in https://github.com/ericpre/miniforge/actions/runs/1090964267 with the different options (admin permission and system install, admin permission and user install, etc.) and it behaves as expected.

For future reference (and because there are a few things entangled) here is a summary of what happens when creating start menu:
1. [packages are installed with `--no-shortcuts` option](https://github.com/conda/constructor/blob/f14fa91e1e363a029fffccb74c81618d27285bfb/constructor/nsis/main.nsi.tmpl#L886)
2. [start menu are created using conda-standalone entry_point](https://github.com/conda/constructor/blob/f14fa91e1e363a029fffccb74c81618d27285bfb/constructor/nsis/main.nsi.tmpl#L904)
3. [the shipped module is used](https://github.com/conda-forge/conda-standalone-feedstock/blob/374ae2c27efcf5fd5cd38ddea316f59ff242067f/recipe/entry_point.py#L78-L87)

With https://github.com/conda/constructor/pull/455/commits/c17eeca95f25e2624b054ee72e09f85251be8325, the menu shortcuts are created _after_ the packages are installed, therefore, it should be possible to remove shipping `menuinst` in `conda-standalone`? The entry point will still be necessary and it would require a release of `menuinst`. @nehaljwani, does it make sense to you?

Reply:
@nehaljwani, can you please check if my explanation is sensible?
If you are happy with it, then we can ping the menuinst maintainer to make a release.